testsuite/openshift/client.py

```diff
@@ -132,14 +132,29 @@ def is_ready(self, selector: Selector):
         success, _, _ = selector.until_all(success_func=lambda obj: "readyReplicas" in obj.model.status)
         return success
 
-    def create_tls_secret(self, name: str, certificate: Certificate, labels: Optional[Dict[str, str]] = None):
-        """Creates a TLS secret"""
+    def create_key_secret(
+        self,
+        name: str,
+        certificate: Certificate,
+        type_: Literal["kubernetes.io/tls", "Opaque"],
+        labels: Optional[Dict[str, str]] = None,
+    ):
```

Comment:
I would add two parameters: key_name, cert_name which we would use as keys for the secret and wouldn't make it bound to the specific type